### PR TITLE
Alerting: visual alert type picker

### PR DESCRIPTION
**Pull request by gillesdemey**
**What this PR does / why we need it**:

Adds a more visual selection for alert types and selects the "Grafana Managed" type by default. Part of an ongoing epic to improve the alert creation flow.

Screenshot 2022-03-17 at 14 39 59

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

The language and inline documentation still has to be revised and signed off on.

### CODE REVIEW

**Comment by konrad147** [comment]
I wonder if we should disable or hide Prometheus-based rule types when there are no Prometheus data sources available. As a second step for the Prometheus-based alert rule, the user needs to select a data source for this rule and when there are no Prometheus-based data sources the list is empty. It seems a bit awkward to give users a choice of options when they are not able to accomplish some of the paths. What do you think?

**Comment by gillesdemey** [comment]
Definitely a great suggestion; the initial focus here was on updating the visuals only but I think it's a good idea to also add that behavior of hiding it since we've been talking about that for a while now :)

**Comment by gillesdemey** [comment]
I've updated the PR to disable the selection of Lotex rules if no compatible datasources are found – for a datasource to be compatible it must:

- be either a Loki or Cortex database with the ruler API enabled
- the data source must have "Manage alerts via Alerting UI" enabled

**Review by konrad147 — APPROVED** [review]
Looks really nice! 🚀  Just one minor nit

File: public/app/features/alerting/unified/components/rule-editor/rule-types/GrafanaManaged.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      image="/public/img/grafana_icon.svg"
+      selected={selected}
+      disabled={disabled}
+      value={RuleFormType.grafana}
+      onClick={onClick}
+    />
+  );
+};
+
+export default GrafanaManagedRuleType;
```
[the remark this answers is not in the document]
Comment (gillesdemey) [reply public/app/features/alerting/unified/components/rule-editor/rule-types/GrafanaManaged.tsx:25]: Yep, good point – I've sinned here 😈

**Comment by achatterjee-grafana** [comment]
Should we be little more clear in the definitions - for example, for Grafana managed alerts, can users select on-prem Prometheus as their data source?

Also, we should be little more clear on what we mean by "Prometheus-style alert". The cloud documentation defines these as "Prometheus-style rules that enable you to query your Grafana Cloud Metrics and then set up Prometheus Alertmanager-style alerts based on those rules. This is useful when you want to create precise, PromQL-based rules or create alerts from across many metrics and logs being collected into your Grafana Cloud Metrics. This form of alerting is much more powerful and configurable, but that comes with some complexity."

Happy to understand more and come up with short& descriptive text strings.

**Comment by achatterjee-grafana** [comment]
Hi @gillesdemey, I wrote up the following. Let me know what you think:

Grafana managed alert - Select this option to create an alerting rule to query any backend data source. Optionally, transform the query results using math, reduce, or other server-side expressions. Alerting rule will be evaluated by Grafana's backend engine.

Prometheus-style alert - Select this option to create an alerting rule to query Loki, Cortex, or other Prometheus compatible data sources. Expressions to transform query results are not supported. Alerting rule will be evaluated by the data source's native alerting engine. Benefits can include faster rule processing and improved performance.

**Comment by achatterjee-grafana** [comment]
@gillesdemey and I had a chat about the UI and the descriptions. Here is a summary of what we discussed. 

- We agreed that the "Prometheus-style" alerting string is confusing and can lead to frustrated users. Especially since for regular Prometheus data sources, users can only create Grafana managed alerts. 

- Even though we are not fans of "Cortex/Loki alerts", it is probably better than Prometheus-style. 

- Users will create "Grafana managed alerts" unless they have Cortex or Ruler API enabled Loki. 
Can we update the UI to mention:
"Select grafana managed alerts unless you have Cortex or a Ruler API enabled Loki data source." (or something similar?) It will give users clear guidance on which option to select.

- Can we hide the recording rule option from the Create rule page since it is only relevant for "Cortex/Loki alerts". 
We show this option only if the users select the Cortex/Loki alerts. Then they will get prompted to either create recording rules or use existing ones?

**Comment by gillesdemey** [comment]
@achatterjee-grafana I've made some final tweaks and updated the screenshot in the original issue advising users to use Grafana Managed alerts when they don't have a Cortex / Loki datasource.

Moving the recording rule selection is currently out of scope for this PR though but we might come back to this at a later point in time to revise the decision tree ✌️

**Comment by jessover9000** [comment]
Hi all, this is looking more and more refined! 👍  I'm wondering about the product strategy aspect of Cortex/Loki alerts though – as I've learned talking to Armand and Dee, using Cortex datasource could fix a lot of scaling issues for people and we recommend using this for Prometheus alerting. So not sure why the benefit isn't currently included here? Could we change the second card text as follows:

- Cortex or Loki alert. Fast, high-performance rule processing for Cortex or Loki data sources. Grafana expressions not supported.

Also do we have to actively tell users "Select Grafana managed" if they don't have a Cortex or Loki data source? Couldn't we change the defaults and
1. just hide the other options if users don't have an appropriate data source? 
2. If people have a Cortex/Loki data source make that card the default?
Or in which use cases would we show that little message in the bottom? If we NEED users to make this choice on their own, we definitely need more guidance in this area because those 2 lines of descriptions are not enough to let people make this decision accurately.

**Comment by konrad147** [comment]
I think there is one important benefit of using Grafana Managed Alerts versus Cortex/Loki alerts. Grafana Managed allows users to use multiple data sources for a single alert rule. Cortex/Loki alerts allow to only query the data source they are defined in.
Maybe we should also communicate that difference to the users?

**Comment by armandgrillet** [comment]
In addition to Jess' feedback, the titles could be more aligned by being:

- Grafana alert rule
- Cortex or Loki alert rule
- Cortex or Loki recording rule

The "rule" part is sort of a repetition but it makes things clearer as there is a difference between an alert rule and an alert instance.

**Comment by jessover9000** [comment]
Great point, Konrad. So the first card could be something like:

- Grafana-managed alert. Multiple data sources of any kind in a single alert. Transform your data with expressions.

**Comment by peterholmberg** [comment]
Mentioning Cortex at all here might be confusing because we are not mentioning anywhere else in Grafana or the Cloud portal.

**Comment by gillesdemey** [comment]
I've incorporated the last round of feedback (and also got the test suite to work again) – check the updated screenshot in the body of the PR.

I think this is the most reasonable way forward and should help novices and still cater to power users who have a Cortex or Loki datasource configured.

**Review by armandgrillet — APPROVED** [review]

**Pull request merged**